backend/uclapi/workspaces/views.py

```diff
@@ -0,0 +1,159 @@
+from base64 import b64decode
+
+from common.decorators import uclapi_protected_endpoint
+from common.helpers import PrettyJsonResponse as JsonResponse
+from common.helpers import RateLimitHttpResponse as HttpResponse
+from rest_framework.decorators import api_view
+
+from .occupeye import BadOccupEyeRequest, OccupEyeApi
+
+
+@api_view(["GET"])
+@uclapi_protected_endpoint(personal_data=False)
+def get_rooms(request, *args, **kwargs):
+    api = OccupEyeApi()
+    response_data = {
+        "ok": True,
+        "rooms": api.get_surveys()
+    }
+    return JsonResponse(
+        response_data,
+        rate_limiting_data=kwargs
+    )
+
+
+@api_view(["GET"])
+@uclapi_protected_endpoint(personal_data=False)
+def get_image(request, *args, **kwargs):
+    try:
+        image_id = request.GET['image_id']
+    except KeyError:
+        response = JsonResponse({
+            "ok": False,
+            "error": "No Image ID provided."
+        }, rate_limiting_data=kwargs)
+        response.status_code = 400
+        return response
+
+    api = OccupEyeApi()
+
+    try:
+        (
+            image_b64,
+            content_type
+        ) = api.get_image(image_id)
+    except BadOccupEyeRequest:
+        response = JsonResponse({
+            "ok": False,
+            "error": (
+                "Either the Image ID you requested does "
+                "not exist, or an internal error occured "
```

Comment:
if it's an internal error we should 500 here, right?

Reply:
Thing is, it wouldn't be an internal error on our part. It would more likely be an issue on Cad-Cap's part, and it's very hard for us to debug that. Do we have a protocol for what to do if a service provider fails?

Reply:
Yeah, but from the user's perspective, the entity that's giving them this data is UCL API, not cad-cap and really they shouldn't be concerned with that. We're here to take those thoughts away from them and make life easier, right?

I feel like we should 500 on cad-caps behalf.

Reply:
Looked this up, I agree now :P https://stackoverflow.com/a/27826514/5297057